resource/resource.go

```diff
@@ -335,7 +335,11 @@ func repodirMatches(dir, owner, repo string) error {
 	right := []string{gu.Host, gu.Owner, gu.Repo}
 	for i, l := range left {
 		r := right[i]
-		if strings.ToLower(l) != strings.ToLower(r) {
+		if !strings.EqualFold(l, r) && i == 2 {
+			return fmt.Errorf("repository name you have set in the source configuration (source:repo) %q doesn't correspond"+
+				" to the parsed git configuration %q. Please correct the repo key value. %w", l, r, errWrongRemote,
+			)
+		} else if !strings.EqualFold(l, r) && i < 2 {
 			return fmt.Errorf("remote: %v: got: %q; want: %q: %w", remote, r, l, errWrongRemote)
 		}
```

Comment:
I would not bother with distinguishing the fields. The important thing is the user has enough info to understand what's wrong. So I would just do:
```suggestion
		if !strings.EqualFold(l, r) {
			return fmt.Errorf("Incompatible git remote and resource configuration.\nGit remote: %q\n"+
				"Resource config: host: github.com, owner: %q, repo: %q", remote, owner, repo,
			)
		}
```

Reply:
The key to decide what to print here is to understand who is the audience:

Myself, the author of Cogito, when I wrote the first version of the message thought it was clear.
Then I stumbled upon it when creating the fly_helper pipeline and could not understand it fully! :-)

So I think it is worthwhile to give full information here and carefully select the words we use, for example "git remote" is confusing for the end user in this context.

Reply:
Is this better? `resource source configuration and git repository are incompatible`
by resource source configuration I mean: 
 - what is setup in source key when declaring a resource in pipeline configuration
by git repository I mean:
- local fetched git repository